### DISCUSSION TITLE
GNU Radio AM Transmitter

### DISCUSSION THREAD

**Discussion by celineirvene** [duggabe/gr-control#1] [Q&A] [answered]
Hi Barry,

So I know this is a bit unorthodox, but I have a question only partially related to this project. My question specifically pertains to - https://wiki.gnuradio.org/index.php/Simulation_example:_AM_transmitter_and_receiver. The AM transmitter simulation example you made.

I am a true novice when it comes to SDRs and GNU Radio, but I wanted to adapt your example to transmit via my HackRF One. The question is 4-part and as follows:

1) If I wanted to use a 44.1kHz wav file instead of an audio source, can I just plug an chug a wav source in its place (and adjust the samp_rate accordingly)?

2) In the article you made a note that "For a real radio transmitter, the output of the Multiply block would be fed to band-pass filters and then to Radio Frequency (RF) hardware". Would the BPF be the same as the one you used in the AM receiver example?

3) For my HackRF I use an osmocom sink to transmit, which takes complex input. When going from the BPF to the osmocom source is it better to use a 'Float to Complex' block or change the BPF to complex output?

4) Lastly, how can I alter the AM transmission frequency so I can hear the wav file on the radio? It seems the transmission range of the HackRF is only from 1MHz to 6GHz, but I would like to verify the signal is properly transmitting.

I know this is strange way of going about this and I don't even know if you'll see this post, but if you do any help is appreciated. Thanks so much for considering.

**Comment by duggabe** [comment, marked as answer]
I have created a flowgraph for an AM transmitter. See https://wiki.gnuradio.org/index.php/File:AM_transmit_2_fg.png

The AM broadcast band in the US is 550kHz to 1.7 MHz. If you keep the transmit gain low and use a very short antenna, you should be OK sending at a clear frequency between 1.0 and 1.7 MHz for short tests. If you are a licensed amateur, then you can use 160 meters or other band of your choice.

**Reply by celineirvene** [reply]
Thank you for this!

I have implemented and tested with my HackRF One, but cannot verify the transmission on the radio between 1MHz and 1.7MHz. I believe this to be a limitation of the HackRF and don't have another SDR to serve as a receiver for now. Any further suggestions? If not I will continue to tweak it and possibly just accept the outcome.

**Reply by duggabe** [reply]
I am not familiar with the HackRF. I would just double-check all the parameters you have set for it. BTW, in my flowgraph I set the transmit frequency to 1.805MHz to be legal in the 160 meter band. Also you can put a GUI Frequency Sink or Time Sink on the output of the Low Pass Filter to see if it looks OK.

**Reply by celineirvene** [reply]
Thanks I will try this!
